compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -290,8 +297,11 @@ object RefChecks {
    *  TODO check that classes are not overridden
    *  TODO This still needs to be cleaned up; the current version is a straight port of what was there
    *       before, but it looks too complicated and method bodies are far too large.
+   *
+   *   @param makeOverridePairsChecker  A function for creating a OverridePairsChecker instance
```

Comment:
I think we might be able to streamline this, maybe by moving checkOverrides into OverridingPairsChecker. But we can do that later after this PR is merged.